### PR TITLE
Fix for #4998

### PR DESCRIPTION
**Pull request by rynowak**
This if the fix for #4998 for `dev` See the commits for details

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs [hunk 1]
```diff
@@ -22,11 +23,9 @@ public class ViewComponentResultExecutor
...
         private readonly IModelMetadataProvider _modelMetadataProvider;
         private readonly ITempDataDictionaryFactory _tempDataDictionaryFactory;
-        private readonly IViewComponentHelper _viewComponentHelper;
 
         public ViewComponentResultExecutor(
             IOptions<MvcViewOptions> mvcHelperOptions,
-            IViewComponentHelper viewComponentHelper,
             ILoggerFactory loggerFactory,
             HtmlEncoder htmlEncoder,
             IModelMetadataProvider modelMetadataProvider,
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs]: This is a `.Internal` class so changes are OK.

File: test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs [hunk 2]
```diff
@@ -33,6 +33,8 @@
 using Newtonsoft.Json.Serialization;
 using Newtonsoft.Json;
 using Xunit;
+using Microsoft.Extensions.ObjectPool;
+using System.Diagnostics;
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs]: will sort

File: src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs [hunk 3]
```diff
@@ -112,8 +105,12 @@ public class ViewComponentResultExecutor
                     writer,
                     _htmlHelperOptions);
 
-                (_viewComponentHelper as IViewContextAware)?.Contextualize(viewContext);
-                var result = await GetViewComponentResult(_viewComponentHelper, _logger, viewComponentResult);
+                // IViewComponentHelper is scoped, we want to make sure to retrieve it every time we need it, as it is
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs]: is ~~scoped~~ transient

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewComponentResultExecutor.cs]: How about I'll just say that it's stateful and leave it at that :laughing:

File: test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs [hunk 4]
```diff
@@ -245,6 +247,32 @@ public void AddMvcCore_AddsMvcJsonOption()
             Assert.Single(services, d => d.ServiceType == typeof(IConfigureOptions<MvcJsonOptions>));
         }
 
+        [Fact]
+        public void AddMvc_NoScopedServiceIsReferredToByASingleton()
+        {
+            var services = new ServiceCollection();
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Test/MvcServiceCollectionExtensionsTest.cs]: > // Arrange, Act, Assert

**Comment by pranavkm** [comment]
:shipit:

**Pull request closed without merging**